lib/pages/login.dart

```diff
@@ -0,0 +1,154 @@
+import 'package:flutter/material.dart';
+import 'package:font_awesome_flutter/font_awesome_flutter.dart';
+
+class LoginPage extends StatelessWidget {
+  const LoginPage({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        minimum: const EdgeInsets.symmetric(horizontal: 32.0),
+        child: Center(
+          child: SingleChildScrollView(
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                Padding(
+                  padding: const EdgeInsets.only(bottom: 40.0),
+                  child: ConstrainedBox(
+                    constraints: const BoxConstraints(maxWidth: 300.0),
+                    child: Column(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: [
+                        Padding(
+                          padding: const EdgeInsets.only(bottom: 6.0),
+                          child: Text(
+                            "Welcome! 👋",
+                            style: Theme.of(context).textTheme.displayMedium,
```

Comment:
14 levels of indentation is ridiculous, even for Flutter. To make this more readable, you could break out methods for each of the major UI elements, like so:
```dart
children: [
  _childOne(context),
  _childTwo(context),
]
```
```dart
Widget _childOne(BuildContext context) => YourWidgetHere();
```
This will make it easier to find and edit specific elements, or reorder them on the page.

Reply:
For example, this page might look something like:
```dart
children: [
  _welcomeText(context),
  _signInWithGoogle(context),
  _signInWithApple(context),
  _orDivider(context),
  _emailField(context), // combining header and text field
  _passwordField(context), // combining header, text field, and 'forgot password'
  _signInButton(context),
  _createAccountText(context),
]
```